tools/scripts/start-local-registry.ts

```diff
@@ -37,70 +49,55 @@ function startLocalRegistry({
+  return new Promise((resolve, reject) => {
+    executeProcess({
+      command: 'npx',
+      args: [
         'nx',
         ...`run ${localRegistryTarget} --location none --clear true`.split(' '),
         ...(storage ? [`--storage`, storage] : []),
       ],
-      { stdio: 'pipe', shell: true },
-    );
-
-    const listener = data => {
-      if (verbose) {
-        process.stdout.write(data);
-      }
-      if (data.toString().includes('http://localhost:')) {
-        const port = parseInt(
-          data.toString().match(/localhost:(?<port>\d+)/)?.groups?.port,
-        );
-        console.info('Local registry started on port ' + port);
+      options: {
+        stdio: 'pipe',
+        shell: true,
+      },
+      observer: {
+        onStdout: (data, childProcess) => {
+          if (verbose) {
+            process.stdout.write(data);
+          }
 
-        const registry = `http://localhost:${port}`;
-        process.env.npm_config_registry = registry;
-        execSync(
-          `npm config set //localhost:${port}/:_authToken "secretVerdaccioToken"`,
-        );
+          if (data.toString().includes('//localhost:')) {
+            const registryData = parseRegistryData(data);
 
-        // yarnv1
-        process.env.YARN_REGISTRY = registry;
-        // yarnv2
-        process.env.YARN_NPM_REGISTRY_SERVER = registry;
-        process.env.YARN_UNSAFE_HTTP_WHITELIST = 'localhost';
+            configureRegistry(registryData);
 
-        console.info('Set npm and yarn config registry to ' + registry);
-
-        resolve(() => {
-          childProcess.kill();
-          execSync(`npm config delete //localhost:${port}/:_authToken`);
-        });
-        childProcess?.stdout?.off('data', listener);
-      }
-    };
-    childProcess?.stdout?.on('data', listener);
-    childProcess?.stderr?.on('data', data => {
-      process.stderr.write(data);
-    });
-    childProcess.on('error', err => {
-      console.error('local registry error', err);
-      reject(err);
-    });
-    childProcess.on('exit', code => {
-      console.info('local registry exit', code);
-      if (code !== 0) {
-        reject(code);
-      } else {
-        resolve(() => {});
+            resolve({
+              registryData,
+              stop: () => {
+                // this makes the process throw
+                childProcess.kill();
+                unconfigureRegistry(registryData);
+                execSync(
+                  `npm config delete ${registryData.registryNoProtocol}/:_authToken`,
+                );
+              },
+            });
+          }
+        },
+        onStderr: data => {
+          if (verbose) {
+            process.stdout.write(data);
+          }
+        },
+        onComplete: code => {
+          console.info('local registry onComplete', code);
+        },
+      },
+    }).catch(error => {
+      if (error.message !== 'Failed to start verdaccio: undefined') {
+        reject(error);
       }
```

Comment:
The promise would remain unfulfilled `if error.message !== 'Failed to start verdaccio: undefined'`.

Reply:
done